lambda_functions/analyzer/main.py

```diff
@@ -155,6 +155,10 @@ def analyze_lambda_handler(event: Dict[str, Any], lambda_context: Any) -> Dict[s
             binary.save_matches_and_alert(
                 lambda_version, os.environ['YARA_MATCHES_DYNAMO_TABLE_NAME'],
                 os.environ['YARA_ALERTS_SNS_TOPIC_ARN'])
+        else:
+            LOGGER.info('%s did not match any YARA rules', binary)
+            binary.safe_alert_only(
+                os.environ['SAFE_SNS_TOPIC_ARN'])
```

Comment:
If safe alerts are disabled, this Lambda function will fail right here, won't it? The `SAFE_SNS_TOPIC_ARN` will be an empty string, and the SNS publish will fail.

The fix might be as simple as:

```python
else:
    LOGGER.info('%s did not match any YARA rules', binary)
    if os.environ['SAFE_SNS_TOPIC_ARN']:
        binary.safe_alert_only(...)
```